### DISCUSSION TITLE
Vertico prevents sudo in Tramp mode

### DISCUSSION THREAD

**Discussion by chrishowejones** [minad/vertico#499] [Q&A]
# Issue

When attempting to use tramp mode to open a protected file (using find-file and then prefixing file path with "/sudo::") the file is not found and the buffer is opened to a new file with the literal path starting with "/sudo::".

E.g.
M-x find-file /sudo::/etc/bluetooth/main.conf

Results in creating a buffer for a new file: "/sudo::/etc/bluetooth/main.conf".

Attempting this with Vertico mode disabled or not required results in the expected behavior i.e. in this example the file at path "/etc/bluetooth/main.conf" is opened using Tramp mode and is writable.

## Minimal config to reproduce

Emacs version: GNU Emacs 29.4 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0) of 2024-06-24
Tramp version: both 2.6.2 or 2.7.1 were tried and issue occurred in both.
Vertico version: 20240629.921

**Comment by minad** [comment]
Hi! I don't see such problems in my setup `/ssh::` or `/su::` on my setup. However I've configured ssh to authenticate automatically via the ssh agent. Such a failure might happen if the connection cannot be established if authentication fails. Vertico accesses the file completion table with the variable `non-essential` set to `t`, which can prevent authentication. Basically the question is - does Tramp ask you for a password when you use `/sudo::/` without Vertico?
